conans/util/windows.py

```diff
@@ -58,8 +61,24 @@ def path_shortener(path, short_paths):
 
     short_home = os.getenv("CONAN_USER_HOME_SHORT")
     if not short_home:
-        drive = os.path.splitdrive(path)[0]
-        short_home = os.path.join(drive, os.sep, ".conan")
+        if OSInfo().is_cygwin:
+            try:
+                cmd = ['cygpath', path, '--unix']
+                out, _ = subprocess.Popen(cmd, stdout=subprocess.PIPE, shell=False).communicate()
+                out = decode_text(out)
+                if out.startswith('/cygdrive/'):  # It was a Windows 'path'
+                    _, _, drive, _ = out.split('/', 3)
+                    short_home = os.path.join('/cygdrive', drive, '.conan')
+                else:  # It was a cygwin path, use a path inside the user home
+                    short_home = os.path.join(os.path.expanduser("~"), '.conan_short')
```

Comment:
Is there a problem to put it always in /cygdrive/c/.conan?

Reply:
I think that, if you are using everything inside Cygwin, you expect everything to be inside Cygwin filesystem... but these are just my expectations.

OTOH, I suppose that `/cygdrive/c` always exists (how common is a Windows without `C` drive?). It could be better if there is a native-windows Conan and its cache is in the `C:` drive...

Do you think it is better to put it in `/cygdrive/c/.conan`? Do we have to check that `C:` exists and fallback to `D`, `E`,...? Do we have to check that there is native-windows Conan and its cache is in the `C` drive?

Reply:
I wouldn't rely on /cygdrive/c existance. it's uncommon, but possible to install Windows on non-C drive.

Reply:
I didn't mean /cygdrive/c/.conan, but ``short_home = os.path.join('/cygdrive', drive, '.conan')`` always. To be consistent with the other short paths default folder (not that I like it, I really dislike it, should be fixed when possible)

Reply:
When it is a Cygwin path, there is no _"drive"_ , the path is something like `/home/<user>/.conan/data/....`.

Reply:
this is true only if you're using Cygwin's $HOME variable to resolve Conan User Home directory. on Windows, you probably will use something different (might be ** SHGetFolderPathW** or whatever python implementation does).
If you will try to use Cygwin's $HOME directory as a cache, then you will have no sharing between Cygwin shell and Windows shells (cmd.exe or PowerShell). so if you are used to running Conan commands in the various shells (like me), you will end up in double disk space usage for things like boost, etc, which is not something good. I think this implementation avoids this, right?

Reply:
Yes, that's true. But sharing the same Conan installation between Windows and Cygwin is not always smooth and maybe it is something we don't support. [Here](https://github.com/conan-io/conan/issues/6751) I summarize some problems and [other users](https://github.com/conan-io/conan/issues/6660#issuecomment-605132351) question it too. 

When I say _"it is a Cygwin path", I mean it is one inside the filesystem of Cygwin and those don't start with `/cygdrive/<drive>/...`. For other paths, it will enter the previous condition in the if-else.

Reply:
I'd say it should be consistent in all cases. it might be surprising on one machine it uses /cygdrive, and on another /home, just because of the different python interpreter or their order in PATH. so yes, I mean, the PATH for user home should start from `/cygdrive`, not from `/home` in case of Cygwin.

Reply:
👍 , that's something for this issue: https://github.com/conan-io/conan/issues/6751